### PR TITLE
Improve section about tags and imagePullPolicy

### PR DESCRIPTION
**Pull request by blacksails**
This is an attempt to make the explaination of tags and
`imagePullPolicy` combinations less confusing.

### CODE REVIEW

**Comment by blacksails** [comment]
I signed it

File: site/content/docs/user/quick-start.md [hunk 1]
```diff
@@ -135,13 +135,12 @@ kubectl apply -f my-manifest-using-my-image:unique-tag
...
 `IfNotPresent` causes the Kubelet to skip pulling an image if it already exists.
-If you want those images loaded into node to work as expected, please:
+If you want those images loaded into node to work as expected, please either:
 
-- don't use a `:latest` tag
-
-and / or:
-
-- specify `imagePullPolicy: IfNotPresent` or `imagePullPolicy: Never` on your container(s).
+- Leave `imagePullPolicy` unset so that it uses the default behavior
```
Comment (BenTheElder) [inline site/content/docs/user/quick-start.md]: The default image pull policy depends on the tag.

File: site/content/docs/user/quick-start.md [hunk 2]
```diff
@@ -135,13 +135,12 @@ kubectl apply -f my-manifest-using-my-image:unique-tag
...
+- If you use a `:latest` tag set `imagePullPolicy` to `Always`
```
Comment (BenTheElder) [inline site/content/docs/user/quick-start.md]: You probably want it to explicitly not be always, always will pull. This explanation is the opposite of how to use `kind load ..` ?

**Comment by BenTheElder** [comment]
This is more or less https://kubernetes.io/docs/concepts/containers/images/#updating-images (which we link to) but flipped for forcing NOT pulling when we've "side-loaded" an image.

File: site/content/docs/user/quick-start.md (hunk 1 above)
Comment (blacksails) [reply site/content/docs/user/quick-start.md]: Changed it to only say that you can only use the default if using unique tags.

**Comment by blacksails** [comment]
@BenTheElder I think I get it now. I changed the PR to my most recent understanding 😅

File: site/content/docs/user/quick-start.md [hunk 3]
```diff
@@ -135,13 +135,11 @@ kubectl apply -f my-manifest-using-my-image:unique-tag
...
-If you want those images loaded into node to work as expected, please:
+If you want those images loaded into node to work as expected, please either:
 
-- don't use a `:latest` tag
-
-and / or:
-
-- specify `imagePullPolicy: IfNotPresent` or `imagePullPolicy: Never` on your container(s).
+- If you use a `:latest` tag set `imagePullPolicy` to `IfNotPresent` or `Never`
+- If you use a unique tag whenever your image changes either leave
```
Comment (BenTheElder) [inline site/content/docs/user/quick-start.md]: Thanks for the PR, but, I'm concerned that this doesn't discouragd `:latest` and conflates dealing with `:latest` and generally dealing with `imagePullPolicy` which should be two distinct but related topics, and more generally covered in the upstream documentation.

We should maybe breifly mention that you don't want `Always` above this, and link to the upstream docs again there.

The upstream docs recommend against latest, and they're right to do so, we should find a way to continue to suggest that here.

The "either or" should probably still be written along the lines of "please not use a :latest tag, or if you must, then use these settings".

**Comment by blacksails** [comment]
I added a sentence mentioning that we would like to avoid the `Always` pull policy, and made it clear that `:latest` is discouraged.

File: site/content/docs/user/quick-start.md [hunk 4]
```diff
@@ -132,16 +132,21 @@ kind load docker-image my-custom-image:unique-tag
...
 
-- don't use a `:latest` tag
+- don't use a `:latest` tag. If you must, then you required to set
+  `imagePullPolicy` to `IfNotPresent` or `Never`.
 
-and / or:
+or
 
-- specify `imagePullPolicy: IfNotPresent` or `imagePullPolicy: Never` on your container(s).
+- use a unique tag whenever your image changes. Then you can leave
```
Comment (BenTheElder) [inline site/content/docs/user/quick-start.md]: you don't need to use a unique tag though, it just needs to not be `:latest`.

**Pull request closed without merging**